helm-framework/helm/provider.go

```diff
@@ -339,9 +338,14 @@ func (p *HelmProvider) Configure(ctx context.Context, req provider.ConfigureRequ
 	if resp.Diagnostics.HasError() {
 		return
 	}
+
+	debug := false
+	if os.Getenv("HELM_DEBUG") == "true" {
+		debug = true
+	}
 	// Override environment variables if the configuration values are provided
```

Comment:
I'm curious how the config overrides the environment variables if both cases set the variable to true.

1. Getting the `HELM_DEBUG` where it's "true" will mark `debug = true
2. Getting the configuration values will mark `debug = true`

Yet i don't see how we apply the override of config values instead of env variables if:
> configuration values are set AND `HELM_DEBUG` envvar is set to "true"

Reply:
the logic sets debug to true if either the environment variable HELM_DEBUG is "true" or if the configuration explicitly sets debug to true. 

with my understanding this ensures that debugging is enabled if either source requests it.

Reply:
In that case we should consider treating the two conditionals as one:
https://github.com/hashicorp/terraform-provider-helm/blob/393479e17b5d30b028690543568dc68c2143060b/helm-framework/helm/provider.go#L342-L348

can be switched to:
```hcl
debug := false
 if os.Getenv("HELM_DEBUG") == "true" || !config.Debug.IsNull() {
 	debug = true 
}
```

Reply:
Gotcha! Due to the previous confusion where I treated two conditions as one, it wasn't very clear what was going on. But these checks are self explanatory so I will update it!

Reply:
At a second glance I'm wondering if it's right to use `!config.Debug.IsNull()`

Since from my understanding if a user sets `debug = false` it is technically not null anymore and would return true and mark `debug = true` despite debug being set to false in the config. Thoughts?

Reply:
Good catch, you are right if the user sets debug. = false, due to it check if its not null it was simply just override it to true. I believe in order to properly handle this scenario > we should check both the null state and the actual boolean value of the Debug field. 

I believe it should look something like this > 
`debug := false

	if os.Getenv("HELM_DEBUG") == "true" {
		debug = true
	}

	if !config.Debug.IsNull() {
		debug = config.Debug.ValueBool()
	}`

Reply:
i believe we can still keep them as one condition since from the docs it says:
>ValueBool returns the known bool value. If Bool is null or unknown, returns false. 

So in other words it would return the "known bool value" so it would return `true` if `debug = true` and `false` if `debug = false` in config.

if the value is `unknown` or `Null` would just return `false`.